src/Diagnostics.RuntimeHost/Startup.cs

```diff
@@ -85,36 +85,39 @@ public void ConfigureServices(IServiceCollection services)
                     {
                         OnTokenValidated = context =>
                         {
+                            context.Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out var requestId);
                             var allowedAppIds = Configuration["SecuritySettings:AllowedAppIds"].Split(",").Select(p => p.Trim()).ToList();
                             var claimPrincipal = context.Principal;
                             var incomingAppId = claimPrincipal.Claims.FirstOrDefault(c => c.Type.Equals("appid", StringComparison.CurrentCultureIgnoreCase));
                             if (incomingAppId == null || !allowedAppIds.Exists(p => p.Equals(incomingAppId.Value, StringComparison.OrdinalIgnoreCase)))
                             {
                                 context.Fail("Unauthorized Request");
-                                DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"AAD Authentication failed because incoming app id was not allowed");
+                                DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"AAD Authentication failed because incoming app id was not allowed, Request id {requestId}");
```

Comment:
Isn't the request id logged in the RequestID column?

Reply:
Oh i see, why not add a new requestid column to the logger? 
https://github.com/Azure/Azure-AppServices-Diagnostics/blob/e305a03d3046a358ec538392d791f995545af8dd/src/Diagnostics.Logger/ETWProvider/DiagnosticsETWProvider.cs#L99-L103

Reply:
This event id is used many places like Services which needs additional code change to send the request id. I did not want to bring too many changes. This is rare cases when AAD fails, and we need to know the reason

Reply:
Ah i see.